content/en/blog/_posts/2023-07-24-mixed-version-proxy-alpha.md

```diff
@@ -0,0 +1,46 @@
+2. To prevent indefinite rerouting of the request, a new header `X-Kubernetes-APIServer-Rerouted:true` is added to the original request once it is determined that the request cannot be served by the original apiserver that received the request and should therefore be proxied. The destination peer apiserver ensures to serve the request locally if it sees this header.
+3. To set the network location of a kube-apiserver that peers will use to proxy requests, the value passed in `--advertise-address` or (when `--advertise-address` is unspecified) the `--bind-address` flag is used. For users with network configurations that would not allow communication between peer kube-apiservers using the addresses specified in these flags, there is an option to pass in the correct peer address as `--peer-advertise-ip` and `--peer-advertise-port` flags that are introduced in this feature.
+
+## How do I enable this feature?
+Following are the required steps to enable the feature:
+
+* Download the latest Kubernetes project (version [v1.28.0](http://v1.28.0/) or later)  
+* Switch on the feature gate with the command line flag `--feature-gates=UnknownVersionInteroperabilityProxy=true` on the kube-apiservers
+* Pass the CA bundle that will be used by source kube-apiserver to authenticate destination kube-apiserver's serving certs using the flag `--peer-ca-file` on the kube-apiservers. Note: this is a required flag for this feature to work. There is no default value enabled for this flag.
+* Pass the correct ip and port of the local kube-apiserver that will be used by peers to connect to this kube-apiserver while proxying a request. Use the flags `--peer-advertise-ip` and `peer-advertise-port` to the kube-apiservers upon startup. If unset, the value passed to either `--advertise-address` or `--bind-address` is used. If those too, are unset, the host's default interface will be used.
+
+## What’s missing?
+Currently we only proxy resource requests to a peer kube-apiserver when its determined to do so. Next we need to address how to work discovery requests in such scenarios. Right now we are planning to have the following capabilities for beta
+
+* Merged discovery across all kube-apiservers
+* Use an egress dialer for network connections made to peer kube-apiservers
+
+## How can I learn more?
+
+- [KEP-4020: Mixed Version Proxy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/4020-unknown-version-interoperability-proxy)
```

Comment:
Even better: link to the documentation first, and the KEP second.

Reply:
Done, but the link will be broken since the documentation is not in the main branch yet?

Reply:
That's not a problem - we never publish post-release blog articles before the release.